homeassistant/components/habitica/__init__.py

```diff
@@ -74,9 +106,47 @@
         vol.Optional(ATTR_TASK): cv.string,
     }
 )
+SERVICE_UPDATE_TASK_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_CONFIG_ENTRY): ConfigEntrySelector(),
+        vol.Required(ATTR_TASK): cv.string,
+        vol.Optional(ATTR_RENAME): cv.string,
+        vol.Optional(ATTR_DESCRIPTION): cv.string,
+        vol.Optional(ATTR_TAG): vol.All(cv.ensure_list, [str]),
+        vol.Optional(ATTR_REMOVE_TAG): vol.All(cv.ensure_list, [str]),
+        vol.Optional(ATTR_ALIAS): vol.All(
+            cv.string, cv.matches_regex("^[a-zA-Z0-9-_]+$")
+        ),
```

Comment:
What are we matching

Reply:
Only alphanumerics, `-`and `_`are allowed for aliases